core/dbt/contracts/graph/model_config.py

```diff
@@ -335,6 +335,39 @@ def replace(self, **kwargs):
 @dataclass
 class SourceConfig(BaseConfig):
     enabled: bool = True
+    quoting: Dict[str, Any] = field(
+        default_factory=dict,
+        metadata=MergeBehavior.Update.meta(),
+    )
+    freshness: Optional[Dict[str, Any]] = field(
+        default=None,
+        metadata=CompareBehavior.Exclude.meta(),
+    )
+    loader: Optional[str] = field(
+        default=None,
+        metadata=CompareBehavior.Exclude.meta(),
+    )
+    # TODO what type is this? docs say: "<column_name_or_expression>"
```

Comment:
String is right.

Food for thought: These dataclass attributes are duplicative with the ones already defined:
- in [UnparsedSourceTableDefinition](https://github.com/dbt-labs/dbt-core/blob/899b0ef2241d97d97df5270c20b67be02ec7ca61/core/dbt/contracts/graph/unparsed.py#L243-L256)
- in [ParsedSourceDefinition](https://github.com/dbt-labs/dbt-core/blob/899b0ef2241d97d97df5270c20b67be02ec7ca61/core/dbt/contracts/graph/parsed.py#L627-L641)

Given our aim here is backwards compatibility, that duplication may be unavoidable